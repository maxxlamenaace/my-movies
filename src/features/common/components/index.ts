import Loader from './loader/Loader';
import Logo from './logo/Logo';
import Footer from './footer/Footer';
import AppBar from './app-bar/AppBar';
import Container from './container/Container';
import ScrollableAppBar from './scrollable-app-bar/ScrollableAppBar';
import Sidebar from './sidebar/Sidebar';
import ManualSwiper from './manual-swiper/ManualSwiper';
import Wrapper from './wrapper/Wrapper';
import PaginatorSwiper from './paginator-swiper/PaginatorSwiper';

export {
  Loader,
  Logo,
  Footer,
  AppBar,
  Container,
  ScrollableAppBar,
  Sidebar,
  ManualSwiper,
  Wrapper,
  PaginatorSwiper,
};
