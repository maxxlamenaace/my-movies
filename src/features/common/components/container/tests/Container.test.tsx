import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { Container } from '@/features/common';

describe('Container', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders correctly', () => {
    const { asFragment } = render(<Container />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('when header is defined', () => {
    it('renders with a header', () => {
      render(<Container header='header' />);

      expect(document.querySelector('#container-header')).toBeDefined();
    });
  });

  describe('when container contains children', () => {
    it('renders with the children', () => {
      render(
        <Container>
          <div className='children' />
        </Container>,
      );

      expect(document.querySelector('#children')).toBeDefined();
    });
  });
});
