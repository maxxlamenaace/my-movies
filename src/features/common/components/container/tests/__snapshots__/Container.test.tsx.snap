// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1cig8sb"
  >
    <div
      class="css-678bp8-MuiStack-root"
    />
  </div>
</DocumentFragment>
`;
